ngraph/core/shape_inference/include/detection_output_shape_inference.hpp

```diff
@@ -0,0 +1,182 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <openvino/op/detection_output.hpp>
+
+namespace ov {
+namespace op {
+namespace v0 {
+
+template <class T>
+void shape_infer(const DetectionOutput* op, const std::vector<T>& input_shapes, std::vector<T>& output_shapes) {
+    using dim_t = typename std::iterator_traits<typename T::iterator>::value_type;
+    NODE_VALIDATION_CHECK(op, (input_shapes.size() == 3 || input_shapes.size() == 5) && output_shapes.size() == 1);
+
+    auto& ret_output_shape = output_shapes[0];
+    ret_output_shape.resize(4);
+
+    const auto& box_logits_pshape = input_shapes[0];
+    const auto& class_preds_pshape = input_shapes[1];
+    const auto& proposals_pshape = input_shapes[2];
+
+    const int num_loc_classes = op->m_attrs.share_location ? 1 : op->m_attrs.num_classes;
+    const int prior_box_size = op->m_attrs.normalized ? 4 : 5;
+
+    dim_t num_images{};
+    dim_t num_prior_boxes{};
+
+    if (box_logits_pshape.rank().is_static()) {
+        NODE_VALIDATION_CHECK(op,
+                              box_logits_pshape.size() == 2,
+                              "Box logits rank must be 2. Got " + std::to_string(box_logits_pshape.size()));
```

Comment:
You can do it as simple as
```suggestion
                              "Box logits rank must be 2. Got ", box_logits_pshape.size());
```